akka-typed/src/main/scala/akka/typed/scaladsl/TimerScheduler.scala

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.scaladsl
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Support for scheduled `self` messages in an actor.
+ * It is used with `Actor.withTimer`, which also takes care of the
+ * lifecycle of the timers such as cancelling them when the actor
+ * is restarted or stopped.
+ */
+trait TimerScheduler[T] {
+
+  /**
+   * Each timer has a key and if a new timer with same key is started
+   * the previous is cancelled and it's guaranteed that a message from the
+   * previous timer is not received, even though it might already be enqueued
+   * in the mailbox when the new timer is started.
+   */
+  def startTimer(key: Any, msg: T, timeout: FiniteDuration, repeat: Boolean): Unit
+
+  /**
+   * Check if a timer with a given `key` is started.
+   */
+  def isTimerActive(key: Any): Boolean
+
+  /**
+   * Cancel a timer with a given `key`. It's guaranteed that a message from the
```

Comment:
I'd rephrase the guarantee a bit, people may not be sure what we mean by previous here.

```
Cancel a timer with a given `key`. 
If canceling a timer that was already canceled, or key never was used to start a timer this operation is a no-op.

It is guaranteed that a message from a canceled timer, including its previous incarnation for the same key,
will not be received by the actor, even though the message might already be enqueued in the mailbox
when cancel is called.
```

?